controllers/flashsystemcluster_controller.go

```diff
@@ -183,18 +183,12 @@ func (r *FlashSystemClusterReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return result, err
 	}
 
-	r.Log.Info("step: init status/conditions of the FlashSystemCluster resource")
-	if instance.Status.Conditions == nil {
+	r.Log.Info("step: reset progressing conditions of the FlashSystemCluster resource")
+	if util.IsStatusConditionFalse(instance.Status.Conditions, odfv1alpha1.ConditionProgressing) {
 		reason := odfv1alpha1.ReasonReconcileInit
-		message := "Initializing flashsystem ODF resources"
+		message := "processing flashsystem ODF resources"
 		util.SetReconcileProgressingCondition(&instance.Status.Conditions, reason, message)
 		instance.Status.Phase = util.PhaseProgressing
-
```

Comment:
why did you remove the update status call?

Reply:
Previous update status call is used for condition initialization which only executes once when CR is created.
Now the `progressing` is set for each round of reconciling, we could leverage status update to only update once at the end of reconcile function in both good path and any error handling.